### ISSUE TITLE
DiskSpaceUsage reports 5-6x the correct amount

### ISSUE THREAD

**Issue by karalabe** [cockroachdb/pebble#2314]
We're charting the metrics from Pebble, and the `DiskSpaceUsage` method for an app reports 467GB, but checking the disk usage via `du` from a shell results in 84GB seen on disk. I could understand some wiggle room, but a 6x multiplier seem to indicate something is very wrong somewhere.

We have the same infra running leveldb too and reporting via the same mechanisms and those numbers match between charts and physical disk usage; so the fault is either in Pebble or our integration / usage of DiskSpaceUsage. Any pointers if it's our fault?

**Comment by holiman** [comment]
Might be totally unrelated.. but I peeked around a bit and found something odd: 

```golang
	metrics.Table.ZombieCount = int64(len(d.mu.versions.zombieTables))
	for _, size := range d.mu.versions.zombieTables {
		metrics.Table.ZombieSize += size
	}

```
Aside from one instance in `metrics_test.go`, this ^ was the one and only place where something wrote to `metrics.Table.ZombieSize`. To be clear: it is never reset back to zero, nor subtracted from.

**Comment by jbowens** [comment]
Thanks for the report. What SHA are you using?

**Comment by jbowens** [comment]
@RaduBerinde I wonder if we missed something in the recent obsolete tables clean up.

**Comment by karalabe** [comment]
> Thanks for the report. What SHA are you using?

`f2e58dc4a0e1`

https://github.com/ethereum/go-ethereum/blob/master/go.mod#L15

**Comment by RaduBerinde** [comment]
> EDIT: Ignore this, the metrics object here is a fresh empty object

I think you're right, this is likely the bug.

**Comment by karalabe** [comment]
TL;DR. Fix seems to work

**Comment by RaduBerinde** [comment]
The fix is merged at 829675f9481128e55a2c0845697008e34c33c9f5

**Comment by karalabe** [comment]
Hmmmmmm, I thought the fix was ObsoletePhysicalSize...

**Comment by karalabe** [comment]
Seems that my above numbers were just a too short of a run (local laptop) to bring out the true issue. I've updated our benchmarkers to the latest pebble code with your fix and I don't see a blowup reported any more. Will let it finish it's thing (2-3h) and close this issue if everything remains stable. Seems to be. Thanks

**Comment by karalabe** [comment]
Can confirm that latest master branch of pebble fixes this issue. Thank you!

**Comment by RaduBerinde** [comment]
Thank you very much!
